### PR TITLE
Users should get stats of trips made in the last X timeframe

### PR DESCRIPTION
**Pull request by emmanuelnwankwo**
#### What does this PR do?
Enables managers and to get detailed statistics of the number of trips they made within a specified timeframe e.g month, week
#### Description of Task to be completed?
- Create a controller method to handle trip request stats
- Implement the service method for searching request db.
- Implement the endpoint for trip request stats.
- Write unit tests for the functionalities
- Document feature
#### How should this be manually tested?
- Clone the repo and cd into the project folder
- Update or copy `.env` file from `.env.sample`
- Checkout to `ft-trips-stats-by-users-167733679` branch and run `npm install`
- - Run `npm run migrate:reset` && `npm run migrate` && `npm run seed` to create tables in your 
 database and seed the database tables with the necessary app configs.
- Run `docker-compose up --build` to build the docker image and spin up all the necessary services
- As an alternative to running the app on docker, you can run `npm run start:dev`
- Launch [http://localhost:3000/api/users/request/stats?start=2019-09-16&end=2019-09-17](http://localhost:3000/api/users/request/stats?start=2019-09-16&end=2019-09-17) on Postman, check image below
image

### CODE REVIEW

File: src/test/user.test.js [hunk 1]
```diff
@@ -89,3 +90,32 @@ describe('GET /users/requests', () => {
     expect(response).to.have.status(404);
   });
 });
+
+describe('POST /users/request/stats', () => {
+  it('should successfully return the number of trip request count', async () => {
+    const queryObj = {
+      startDate: formatDate(Date.now()),
```
Comment (Baystef) [inline src/test/user.test.js]: Please explain what's happening here, isn't startdate === endDate here?

**Review by Baystef — COMMENTED** [review]
Please check my comment, thanks

**Comment by Baystef** [comment]
Your build is failing, please fix this

**Comment by dchima** [comment]
> Your build is failing, please fix this

@Baystef  The build actually passes. remember the travis-ci.org, travis-ci.com thing. I've removed one from the repo now.

File: src/validation/tripRequestValidation.js [hunk 2]
```diff
@@ -147,6 +147,32 @@ export default class TripRequestValidation {
...
+      startDate: Joi.date()
+        .format('YYYY-MM-DD')
+        .required()
+        .error(TripRequestValidation.validateTripStats('startDate')),
+      endDate: Joi.date()
+        .format('YYYY-MM-DD')
+        .min(body.startDate)
+        .required()
+        .error(TripRequestValidation.validateTripStats('endDate'))
+    };
```
Comment (dchima) [inline src/validation/tripRequestValidation.js:168]: Can you please shed some light on why your validations are written this way?

**Comment by dchima** [comment]
[Users should get stats of trips made in the last X timeframe](https://trello.com/c/1Gwnm8RY/69-users-should-get-stats-of-trips-made-in-the-last-x-timeframe)

File: src/test/user.test.js (hunk 1 above)
Comment (emmanuelnwankwo) [reply src/test/user.test.js]: You're right @Baystef, I forgot to update the test after I made some changes in the validation

File: src/validation/tripRequestValidation.js (hunk 2 above)
Comment (emmanuelnwankwo) [reply src/validation/tripRequestValidation.js:168]: Apart from validating a specific date format, this validation ensures that the **endDate** is not less than **startDate** before querying the database

**Review by dchima — APPROVED** [review]

**Review by Baystef — APPROVED** [review]
Great job bruv

**Review by King-Success — APPROVED** [review]
Good job bro

**Pull request merged**